test/minikube/minikube_base_tde_test.go

```diff
@@ -0,0 +1,225 @@
+	}, 30*time.Second)
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+	databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &options)
+
+	// Generate diagnose in case this test fails
+	testlib.AddTeardown(testlib.TEARDOWN_DATABASE, func() {
+		if t.Failed() {
+			testlib.GetDiagnoseOnTestFailure(t, namespaceName, admin0)
+			testlib.RecoverCoresFromEngine(t, namespaceName, "te", "demo-log-te-volume")
+		}
+	})
+
+	// Enable TDE on database level
+	output, _ := testlib.RunSQL(t, namespaceName, admin0, "demo", "alter database change encryption type AES128")
+	assert.False(t, strings.Contains(output, "Error"), "Failed to enable TDE: %s", output)
+
+	// populate some data
+	k8s.RunKubectl(t, kubectlOptions,
+		"exec", admin0, "--",
+		"/opt/nuodb/bin/nuosql",
+		"--user", "dba",
+		"--password", "secret",
+		"demo",
+		"--file", "/opt/nuodb/samples/quickstart/sql/create-db.sql",
+	)
+
+	// verify that the database contains the populated data
+	tables, _ := testlib.RunSQL(t, namespaceName, admin0, "demo", "show schema User")
+	require.True(t, strings.Contains(tables, "HOCKEY"), "tables returned: %s", tables)
+
+	opt := testlib.GetExtractedOptions(&options)
+	tePodNameTemplate := fmt.Sprintf("te-%s-nuodb-%s-%s", databaseChartName, opt.ClusterName, opt.DbName)
+	smPodNameTemplate := fmt.Sprintf("sm-%s-nuodb-%s-%s", databaseChartName, opt.ClusterName, opt.DbName)
+
+	tePodName := testlib.GetPodName(t, namespaceName, tePodNameTemplate)
+	go testlib.GetAppLog(t, namespaceName, tePodName, "_pre-restart", &corev1.PodLogOptions{Follow: true})
+
+	smPodName0 := testlib.GetPodName(t, namespaceName, smPodNameTemplate)
+	go testlib.GetAppLog(t, namespaceName, smPodName0, "_pre-restart", &corev1.PodLogOptions{Follow: true})
+
+	// Take encrypted backup
+	backupset := testlib.BackupDatabase(t, namespaceName, smPodName0, opt.DbName, "full", opt.ClusterName)
+	options.SetValues["restore.source"] = backupset
+
+	// Drop USER.HOCKEY table before restore
+	testlib.RunSQL(t, namespaceName, admin0, "demo", "drop table USER.HOCKEY")
+	// restore database
+	defer testlib.Teardown(testlib.TEARDOWN_RESTORE)
+	testlib.RestoreDatabase(t, namespaceName, admin0, &options)
+	// verify that the database contains USER.HOCKEY table AFTER the restore
+	tables, _ = testlib.RunSQL(t, namespaceName, admin0, "demo", "show schema User")
+	require.True(t, strings.Contains(tables, "HOCKEY"), "Show schema returned: %s", tables)
+
+	// Perform TDE password rotation
+	passwordNew := strings.ToLower(random.UniqueId())
+	applyStoragePasswordSecret(t, namespaceName, options.SetValues["admin.tde.secrets.demo"], []string{passwordNew, password})
+	testlib.Await(t, func() bool {
+		err := k8s.RunKubectlE(t, kubectlOptions,
+			"exec", admin0, "--",
+			"nuocmd", "check", "data-encryption",
+			"--db-name", opt.DbName,
+			"--password", passwordNew,
+		)
+		return err == nil
+	}, 90*time.Second)
```

Comment:
I notice that this is longer than the 30 seconds you use elsewhere. Is because of the time it takes K8s to update the secret?

Reply:
Exactly --- it takes ~60sec for Kubernetes to update the secret inside the pods. Once the files mounted from the secret are updated, the resync will be triggered almost instantly because of the IO notifications.